packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,243 @@
+import { BrokenImageRichIcon } from "@hopper-ui/icons";
+import { type ResponsiveProp, useStyledSystem, type StyledSystemProps, useResponsiveValue } from "@hopper-ui/styled-system";
+import { filterDOMProps } from "@react-aria/utils";
+import clsx from "clsx";
+import { type CSSProperties, forwardRef, type ForwardedRef, useMemo, type ReactNode, type ReactEventHandler } from "react";
+import { useContextProps } from "react-aria-components";
+
+import { Text, type TextProps } from "../../typography/Text/index.ts";
+import { type SizeAdapter, cssModule, type BaseComponentProps } from "../../utils/index.ts";
+
+import { AvatarContext } from "./AvatarContext.ts";
+import { RichIconAvatarImage } from "./RichIconAvatarImage.tsx";
+import { useImageFallback } from "./useImageFallback.ts";
+
+import styles from "./Avatar.module.css";
+
+export const GlobalAvatarCssSelector = "hop-Avatar";
+export type AvatarSize = "xs" | "sm" | "md" | "lg" | "xl" | "2xl";
+
+type AvatarImageBaseProps = React.HTMLProps<HTMLImageElement>;
+
+export interface AvatarInitialsProps {
+    /**
+     * The name of the Avatar. This will be used for the alt text of the image or the initials if no image is provided.
+     */
+    name: string;
+    /**
+     * The size of the avatar.
+     * * @default "md"
+     */
+    size: ResponsiveProp<AvatarSize>;
+}
+
+export interface AvatarImageProps {
+    children?: ReactNode;
+    /**
+     * Props to add to the img element when src is provided.
+     */
+    imageProps?: AvatarImageBaseProps;
+    /**
+     * Whether or not the avatar is disabled.
+     */
+    isDisabled?: boolean;
+    /**
+     * The size of the avatar.
+     * * @default "md"
+     */
+    size: ResponsiveProp<AvatarSize>;
+    /**
+     * The src of the image to display. If not provided, the initials will be displayed instead.
+     */
+    src: string;
+    /**
+    * The src of the image to display if the image fails to load. If set to null, the initials will be displayed instead.
+    * * @default "BrokenImageRichIcon"
+    */
+    fallbackSrc?: string | null;
+}
+
+export interface AvatarProps extends StyledSystemProps, BaseComponentProps, Omit<AvatarImageProps, "size" | "src">, Omit<AvatarInitialsProps, "size"> {
```

Comment:
i know i'm the one that suggested having AvatarImageProps,AvatarInitialsProps and AvatarProps that extends the 2, but seeing how the props documentation refers each other and how you have to omit stuff in Avatar, i'm thinking i don't like it anymore. 


Have the avatar interface be standlone: 
```
export interface AvatarProps extends StyledSystemProps, BaseComponentProps {
    /**
     * The name of the Avatar. This will be used for the alt text of the image or the initials if no image is provided.
     */
    name: string;

    children?: ReactNode;
    /**
     * Props to add to the img element when src is provided.
     */
    imageProps?: AvatarImageBaseProps;
    /**
     * Whether or not the avatar is disabled.
     */
    isDisabled?: boolean;
    /**
     * The size of the avatar.
     * * @default "md"
     */
    size: ResponsiveProp<AvatarSize>;
    /**
     * The src of the image to display. If not provided, the initials will be displayed instead.
     */
    src?: string;
    /**
    * The src of the image to display if the image fails to load. If set to null, the initials will be displayed instead.
    * * @default "BrokenImageRichIcon"
    */
    fallbackSrc?: string | null;
}
```

And move the AvatarImageProps and Initials interfaces next to their component and duplicate the props needed